sale_order_lot_selection/model/sale.py

```diff
@@ -72,6 +72,7 @@ def _check_move_state(self, line):
     @api.multi
     def action_confirm(self):
         res = super(SaleOrder, self).action_confirm()
```

Comment:
Would have been a shorter diff to write:

```python
if not res:
    return res
```

And avoided an indentation level as well.

Reply:
Actually what is the reason to check `res`? It seems safer to me to actually check the `state` of the sales orders and call `_check_move_state` only if it's the correct state.

Reply:
in `sale-workflow/sale_exception/models/sale.py`
```
    @api.multi
    def action_confirm(self):
        if self.detect_exceptions():
            return self._popup_exceptions()
        else:
            return super(SaleOrder, self).action_confirm()
```
in some case return action `dict` and don't confirm SO.
if SO not in correct state it can cause othe error that overwrite `res` results by `raise` some exception

Reply:
about
```
if not res:
    return res
```
we need to check if `res` contain `True` (is the instance of `bool` and  value is `True`)
```
z = {'a': 1}
if z: print(bool(z))
True
if not z: print(bool(z))

```
not empty `dict` evaluate to `True`
so it not what we want.
we can do something like this
```
if instanceof(res, dict):
    return res
```
but it also not good idea